libraries-3/src/main/java/com/baeldung/barcodes/BarcodesController.java

```diff
@@ -0,0 +1,116 @@
+        BufferedImage image = BarbecueBarcodeGenerator.generateUPCABarcodeImage(barcodeText);
+        return new ResponseEntity<>(image, HttpStatus.OK);
+    }
+
+    @GetMapping(value = "/barbecue/ean13", produces = MediaType.IMAGE_PNG_VALUE)
+    public ResponseEntity<BufferedImage> barbecueEAN13Barcode(@RequestParam("barcodeText") String barcodeText) throws BarcodeException, OutputException {
+        BufferedImage image = BarbecueBarcodeGenerator.generateEAN13BarcodeImage(barcodeText);
+        return new ResponseEntity<>(image, HttpStatus.OK);
+    }
+
+    @GetMapping(value = "/barbecue/code128", produces = MediaType.IMAGE_PNG_VALUE)
+    public ResponseEntity<BufferedImage> barbecueCode128Barcode(@RequestParam("barcodeText") String barcodeText) throws BarcodeException, OutputException {
+        BufferedImage image = BarbecueBarcodeGenerator.generateCode128BarcodeImage(barcodeText);
+        return new ResponseEntity<>(image, HttpStatus.OK);
+    }
+
+    @GetMapping(value = "/barbecue/pdf417", produces = MediaType.IMAGE_PNG_VALUE)
+    public ResponseEntity<BufferedImage> barbecuePDF417Barcode(@RequestParam("barcodeText") String barcodeText) throws BarcodeException, OutputException {
+        BufferedImage image = BarbecueBarcodeGenerator.generatePDF417BarcodeImage(barcodeText);
+        return new ResponseEntity<>(image, HttpStatus.OK);
+    }
+
+    //Barcode4j library
+
+    @GetMapping(value = "/barcode4j/upca", produces = MediaType.IMAGE_PNG_VALUE)
+    public ResponseEntity<BufferedImage> barcode4jUPCABarcode(@RequestParam("barcodeText") String barcodeText) {
+        BufferedImage image = Barcode4jBarcodeGenerator.generateUPCABarcodeImage(barcodeText);
+        return new ResponseEntity<>(image, HttpStatus.OK);
+    }
+
+    @GetMapping(value = "/barcode4j/ean13", produces = MediaType.IMAGE_PNG_VALUE)
+    public ResponseEntity<BufferedImage> barcode4jEAN13Barcode(@RequestParam("barcodeText") String barcodeText) {
+        BufferedImage image = Barcode4jBarcodeGenerator.generateEAN13BarcodeImage(barcodeText);
+        return new ResponseEntity<>(image, HttpStatus.OK);
+    }
+
+    @GetMapping(value = "/barcode4j/code128", produces = MediaType.IMAGE_PNG_VALUE)
+    public ResponseEntity<BufferedImage> barcode4jCode128Barcode(@RequestParam("barcodeText") String barcodeText) {
+        BufferedImage image = Barcode4jBarcodeGenerator.generateCode128BarcodeImage(barcodeText);
+        return new ResponseEntity<>(image, HttpStatus.OK);
+    }
+
+    @GetMapping(value = "/barcode4j/pdf417", produces = MediaType.IMAGE_PNG_VALUE)
+    public ResponseEntity<BufferedImage> barcode4jPDF417Barcode(@RequestParam("barcodeText") String barcodeText) {
+        BufferedImage image = Barcode4jBarcodeGenerator.generatePDF417BarcodeImage(barcodeText);
+        return new ResponseEntity<>(image, HttpStatus.OK);
```

Comment:
It seems like you might be able to create a function to build the response entity

```java
private static ResponseEntity<BufferedImage> okResponse(BufferedImage image) {
    return new ResponseEntity<>(image, HttpStatus.OK);
}
```

This would allow callsites to be a little cleaner:

```java
   return okResponse(generatePDF417BarcodeImage(barcodeText));
```

Reply:
done